### PR TITLE
chore(dep): Unpin slackclient dependency

### PR DESCRIPTION
**Pull request by ecederstrand**
SUMMARY

This is the last of three PRs, attempting to unpin Superset dependencies. See also https://github.com/apache/superset/pull/21581 and https://github.com/apache/superset/pull/21091

`slackclient` was originally pinned in https://github.com/apache/superset/pull/10345 due to https://github.com/slackapi/python-slack-sdk/issues/728 which was fixed in a series of PRs concluding with https://github.com/slackapi/python-slack-sdk/issues/901 which was released in `slack_sdk` version 3.1.1.

`slack_sdk` no longer depends on `aiohttp`, which removes a handful of dependencies.

While here, make `tabulate` and `holidays` dependency versions open-ended. This leaves only `numpy` pinned in `install_requires` in setup.py.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -114,11 +114,11 @@ def get_git_sha() -> str:
         "redis",
         "selenium>=3.141.0",
         "simplejson>=3.15.0",
-        "slackclient==2.5.0",  # PINNED! slack changes file upload api in the future versions
+        "slack_sdk>=3.1.1",
```
Comment (villebro) [inline setup.py:117]: Can we add a `,4` here, just in case?
```suggestion
        "slack_sdk>=3.1.1, <4",
```

File: setup.py [hunk 2]
```diff
@@ -114,11 +114,11 @@ def get_git_sha() -> str:
...
         "sqlalchemy>=1.4, <2",
         "sqlalchemy-utils>=0.38.3, <0.39",
         "sqlparse==0.3.0",  # PINNED! see https://github.com/andialbrecht/sqlparse/issues/562
-        "tabulate==0.8.9",
+        "tabulate>=0.8.9",
```
Comment (villebro) [inline setup.py:121]: ..and here something similar, assuming typical semver?
```suggestion
        "tabulate>=0.8.9, <0.9",
```

**Review by villebro — COMMENTED** [review]
Awesome work, that stale Slack library had started to annoy me, too 🙂 Left a few comments, LMKWYT.

**Comment by villebro** [comment]
Tagging @dpgaspar and @bkyryliuk for additional reviews to validate for potential regressions caused by the Slack client major bump.

File: setup.py [hunk 3]
```diff
@@ -94,7 +94,7 @@ def get_git_sha() -> str:
         "graphlib-backport",
         "gunicorn>=20.1.0",
         "hashids>=1.3.1, <2",
-        "holidays==0.14.2",
+        "holidays>=0.14.2",
```
Comment (villebro) [inline setup.py:97]: While we're at it, should we bring this one up to the current version also?
```suggestion
        "holidays>=0.16.0, <0.17",
```
The changelog looks pretty straight forward, and I kind of like the added support for 3.11 in 0.15 which we'll soon start working towards: https://github.com/dr-prodigy/python-holidays/blob/master/CHANGES

**Review by villebro — APPROVED** [review]
LGTM with one last optional comment

**Review by dpgaspar — APPROVED** [review]
Verified that charts are sent to Slack with the new sdk

**Pull request merged**